bn-expansions/src/calculateWeight.ts

```diff
@@ -0,0 +1,53 @@
+import { BaseCatch } from '@boatnet/bn-models';
+
+export function updateCatchWeight(weightMethod: number, catchVal: BaseCatch) {
+    if (weightMethod === 3) {
+        // TODO
+    } else if (weightMethod === 5) {
+        // TODO
+    } else if (weightMethod === 6) {
+        // TODO
+    } else if (weightMethod === 7) {
+        // TODO
+    } else if (weightMethod === 8) {
+        // TODO
+    } else if (weightMethod === 9) {
+        // TODO
+    } else if (weightMethod === 14) {
+        // TODO
+    } else if (weightMethod === 15) {
+        // TODO
+    } else if (weightMethod === 19) {
+        // TODO
+    } else if (weightMethod === 20) {
+        return getWholeWeight(catchVal);
+    } else if (weightMethod === 21) {
+        // TODO
+    }
+}
+
+// updates baskets and species weights
+function getWholeWeight(catchVal: BaseCatch) {
```

Comment:
Looks good to me!
I think we will want to test this via some automated tests, to verify.
Take a look at boatnet-module/test-boatnet-module/tests for examples of this. After you yarn install, yarn test:unit to see how they work.
Maybe we could add a testdata directory with .json files (stripped of any PII) with stuff to test.

Reply:
okay added a unit test